Public/Src/Engine/UnitTests/Processes/SandboxedProcessBreakawayTest.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using BuildXL.Utilities;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using BuildXL.Processes;
+using Test.BuildXL.Executables.TestProcess;
+using Test.BuildXL.TestUtilities.Xunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Test.BuildXL.Processes
+{
+    public sealed class SandboxedProcessBreakawayTest : SandboxedProcessTestBase
+    {
+        public SandboxedProcessBreakawayTest(ITestOutputHelper output)
+            : base(output) { }
+
+        [TheoryIfSupported(requiresWindowsBasedOperatingSystem: true)]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ChildProcessCanBreakawayWhenConfigured(bool letInfiniteWaiterSurvive)
+        {
+            // We use InfiniteWaiter (a process that waits forever) as a long-living process that we can actually check it can
+            // escape the job object
+            var fam = new FileAccessManifest(
+                Context.PathTable, 
+                childProcessesToBreakawayFromSandbox: letInfiniteWaiterSurvive? new[] { InfiniteWaiterToolName } : null);
+            
+            fam.FailUnexpectedFileAccesses = false;
+
+            // We instruct the regular test process to spawn InfiniteWaiter as a child
+            var info = ToProcessInfo(
+                ToProcess(
+                    Operation.SpawnExe(
+                        Context.PathTable, 
+                        CreateFileArtifactWithName(InfiniteWaiterToolName, TestDeploymentDir))),
+                fileAccessManifest: fam);
+
+            // Let's shorten the default time to wait for nested processes, since we are spawning 
+            // a process that never ends and we don't want this test to wait for that long
+            info.NestedProcessTerminationTimeout = TimeSpan.FromMilliseconds(10);

```

Comment:
I expected all these breakaway processes to be automatically added to `allowedSurvivingChildProcessNames`; when those are specified, after the main process exits SandboxedProcess doesn't for those surviving child processes to exit.

Reply:
Never mind, I see that there is a case when the test doesn't let the infinite waiter survive, in which case there is no pointing waiting longer than 10ms.